### PR TITLE
py: bugfixes from Coverity

### PR DESCRIPTION
**Pull request by prusnak**
Recently, I've been trying Coverity for my projects and I tried it also for MicroPython. There were some interesting findings (around 35) and I'd like to know if you are interested in processing them and what would be the best way how to do it.

I see two options:

a) You go directly to https://scan.coverity.com/projects/micropython/view_defects and I'll give you admin access (needs login with Github ...). We could assess the issues, mark false positives, etc. there.

b) I start preparing pull requests and we'll discuss the changes here in comments of pull requests.

a) puts more effort on your side, b) on mine side.

Included in this PR are two examples of trivial findings. Others would probably need more discussion.

Tell me what you think. Also are you interested in removing asserts that are always true? (second commit of this PR).

### CODE REVIEW

**Comment by pfalcon** [comment]
@prusnak : To give some quick reply, we'd appreciate any reports of real bugs. Example of a real bug: https://github.com/micropython/micropython/issues/2451 . Sorry, but I'd say that examples you gave are more of "artificial". For example mpz_as_str_inpl() is unlikely ever called with NULL. Of course, those thinkos needs to be fixed, so thanks for reporting it, and "py: fix null pointer dereference in mpz.c, fix missing va_end in warning.c" looks good to go, I'll leave it for couple more days for @dpgeorge to look though.

Assert fixes are arguably not correct at all - someone who does them assumes that data types are correct, and asserts are "superfluous", but what if data type changes? Asserts are also kind of documentation.

Summing up - if you find a "real" bug, please share it ASAP, everybody interested to see an automated tool capable of something more than false positives and trivial cases ;-). Otherwise, if you could spool patches for definitive issues with background priority, that would be a good approach in the meantime. Why do you suggest a) specifically? MicroPython could have own account there, though it all boils down to time, that's why I suggest that non-questionable patches is the best way currently indeed.

Thanks, and feel free to wait for @dpgeorge's reply too.

**Comment by dpgeorge** [comment]
Thanks for the work on this.  I'm happy with the patch that fixes the null pointer and va_end.  IMO the assert patch is also fine to merge (ie to remove the always-true assertions) because most of those assertions were written when the argument type was signed (in fact, they were commited in the very first commit!), and then it changed to unsigned without adjusting/removing the assertions.

If you want to continue making small PRs like this that would be the best approach.

**Comment by prusnak** [comment]
OK. Feel free to merge one or both commits from this pull request. I don't have strong opinion about the assert one, that's why I put it here to discuss it.

I will create another pull requests for another issues from Coverity.

**Comment by pfalcon** [comment]
Merged, thanks!
